### PR TITLE
created branch for sync

### PR DESCRIPTION
**Pull request by valeriadamante**

### CODE REVIEW

**Review by kandrosov — CHANGES_REQUESTED** [review]
- update ele selection
- put tau_eta cut in the config
- fix MET

File: Common/AnalysisTools.h [hunk 1]
```diff
@@ -154,8 +138,12 @@ RVecB RemoveOverlaps(const RVecLV& obj_p4, const RVecB& pre_sel, const std::vect
     for(const auto& other_obj_col : other_objects) {
       for(const auto& other_obj_p4 : other_obj_col) {
         const double dR2 = ROOT::Math::VectorUtil::DeltaR2(p4, other_obj_p4);
+        //std::cout << "Jet pt = " << p4.pt() << " tau pt " << other_obj_p4.pt() << " delta R 2 " << dR2 << " dr2min " << min_deltaR2 << std::endl;
+        //std::cout << "Jet eta = " << p4.eta() << " tau eta " << other_obj_p4.eta() << " delta R 2 " << dR2 << " dr2min " << min_deltaR2 << std::endl;
+        //std::cout << std::endl;
```
Comment (kandrosov) [inline Common/AnalysisTools.h:143]:
```suggestion
```

File: Common/AnalysisTools.h [hunk 2]
```diff
@@ -154,8 +138,12 @@ RVecB RemoveOverlaps(const RVecLV& obj_p4, const RVecB& pre_sel, const std::vect
...
         if(dR2 > min_deltaR2) {
           ++cnt;
+          //std::cout << " count " << cnt << std::endl;
```
Comment (kandrosov) [inline Common/AnalysisTools.h:146]:
```suggestion
```

File: Common/BaselineRecoSelection.h [hunk 3]
```diff
@@ -32,6 +32,11 @@ ROOT::VecOps::RVec<HTTCand> GetHTTCandidates(Channel channel, double dR_thr,
         cand.leg_rawIso[1] = leg2_rawIso.at(leg2_idx);
         cand.leg_genMatchIdx[0] = leg1_genMatchIdx.at(leg1_idx);
         cand.leg_genMatchIdx[1] = leg2_genMatchIdx.at(leg2_idx);
+        //std::cout << "leg 1 idx " << leg1_idx << " leg 2 idx " << leg2_idx << std::endl;
+        //std::cout << "idx 1 iso " << cand.leg_rawIso[0] << " idx 2 iso " << cand.leg_rawIso[1] << std::endl;
+        //std::cout << "idx 1 pt " << cand.leg_p4[0].pt() << " idx 2 pt " << cand.leg_p4[1].pt() << std::endl;
+        //std::cout << "idx 1 eta " << cand.leg_p4[0].eta() << " idx 2 eta " << cand.leg_p4[1].eta() << std::endl;
+        //std::cout << std::endl;
```
Comment (kandrosov) [inline Common/BaselineRecoSelection.h:39]:
```suggestion
```

File: Common/BaselineSelection.py [hunk 4]
```diff
@@ -126,15 +125,14 @@ def DefineMETCuts(met_thr, met_collections):
 def RecoLeptonsSelection(df, apply_filter=True):
     df = df.Define("Electron_B0", f"""
         v_ops::pt(Electron_p4) > 18 && abs(v_ops::eta(Electron_p4)) < 2.3 && abs(Electron_dz) < 0.2 && abs(Electron_dxy) < 0.045
-        && (Electron_mvaIso_WP90 || (Electron_mvaNoIso_WP90 && Electron_pfRelIso03_all < 0.5))
-    """)
+    """) # && (Electron_mvaIso_WP90 || (Electron_mvaNoIso_WP90 && Electron_pfRelIso03_all < 0.5))
```
Comment (kandrosov) [inline Common/BaselineSelection.py:128]: don't change any B0 selections!

File: config/samples_Run2_2018.yaml [hunk 5]
```diff
@@ -31,11 +29,12 @@ GLOBAL:
       VSjet: Medium
     tauTau:
       VSe: VVLoose
-      VSmu: Tight
+      VSmu: VLoose
```
Comment (kandrosov) [inline config/samples_Run2_2018.yaml:32]: should be `Tight` isn't it?

**Review by kandrosov — CHANGES_REQUESTED** [review]

File: Common/AnalysisTools.h (hunk 1 above)
Comment (valeriadamante) [reply Common/AnalysisTools.h:143]: ok

File: Common/AnalysisTools.h (hunk 2 above)
Comment (valeriadamante) [reply Common/AnalysisTools.h:146]: ok

File: Common/BaselineRecoSelection.h (hunk 3 above)
Comment (valeriadamante) [reply Common/BaselineRecoSelection.h:39]: ok

File: Common/BaselineSelection.py (hunk 4 above)
Comment (valeriadamante) [reply Common/BaselineSelection.py:128]: ok

File: config/samples_Run2_2018.yaml (hunk 5 above)
Comment (valeriadamante) [reply config/samples_Run2_2018.yaml:32]: nmmh maybe yes. Changed

Comment (valeriadamante) [reply config/samples_Run2_2018.yaml:32]: In this case all WPs are equal:
```
    eTau:
      VSe: VVLoose
      VSmu: Tight
      VSjet: Medium
    muTau:
      VSe: VVLoose
      VSmu: Tight
      VSjet: Medium
    tauTau:
      VSe: VVLoose
      VSmu: Tight
      VSjet: Medium
      ```

**Review by kandrosov — APPROVED** [review]

**Pull request merged**